README.md

```diff
@@ -1,24 +1,272 @@
+- [Redis](https://pypi.org/project/redis/)
+- [Requests](https://pypi.org/project/requests/)
+- [urllib3](https://pypi.org/project/urllib3/)
+
+## Installation
+
+```console
+$ pip install corva-sdk
+```
+
+## App types
+
+There are three app types, that you can build: `stream`, `scheduled` and `task`.<br>
+**Note**: it is recommended to use type hints like in examples below,
+so that editors and tools can give you better support.
+
+#### Stream
+
+```python
+from corva import Api, Cache, Corva, StreamEvent
+
+
+# 1 define a function with required parameters, that will be provided by sdk
+def stream_app(event: StreamEvent, api: Api, cache: Cache):
+    """Main logic function"""
+    pass
+
+
+# 2 define a function that will be run by AWS lambda
+def lambda_handler(event, context):
+    """AWS lambda handler"""
+    corva = Corva()  # 3 initialize Corva
+    corva.stream(stream_app, event)  # 4 run stream_app by passing it and event to Corva.stream
+```
+
+`Corva.stream` provides two additional parameters:
+
+- `filter_by_timestamp` - if set to `True` will take the latest processed
+  `timestamp` from cache and filter out records from event with 
+  either smaller or same `timestamp`;
+- `filter_by_depth` - if set to `True` will take the latest processed
+  `measured_depth` from cache and filter out records from event with
+  either smaller or same `measured_depth`.
+
+#### Scheduled
+
+```python
+from corva import Api, Cache, Corva, ScheduledEvent
+
+
+def scheduled_app(event: ScheduledEvent, api: Api, cache: Cache):
+    pass
+
+
+def lambda_handler(event, context):
+    corva = Corva()
+    corva.scheduled(scheduled_app, event)
+```
+
+#### Task (will be added soon)
+
+```python
+from corva import Api, Corva, TaskEvent
+
+
+# note, that task app doesn't receive cache parameter
+def task_app(event: TaskEvent, api: Api):
+    pass
+
 
-### Set up the project
-```sh
-cd ~/YOUR_PATH/python-worker-public
-python3 -m venv venv
-source venv/bin/activate
-pip install -r requirements.txt
+def lambda_handler(event, context):
+    corva = Corva()
+    corva.task(task_app, event)
 ```
 
-### Run tests
-```sh
-venv/bin/python3 -m pytest tests
+## Event
+
+Event is what triggers app execution.
+It contains necessary data for app to run e.g. `asset_id`, that triggered the event.
+Every app type receives `Event` instance as a first parameter.
+Each app type has its own event type: `StreamEvent`, `ScheduledEvent` and `TaskEvent`. <br>
+**Note**: It is recommended to use event classes as type hints,
+so that editors and tools can give you better support.
+
+#### Example:
+
+```python
+from corva import Api, Cache, Corva, StreamEvent
+
+
+def stream_app(event: StreamEvent, api: Api, cache: Cache):
+    event.records[0].measured_depth  # get measured depth of first event record
+
+
+def lambda_handler(event, context):
+    corva = Corva()
+    corva.stream(stream_app, event)
+```
+
+## Api
+
+The apps might need to communicate with Corva API and Corva Data API.
```

Comment:
does the user need to know the difference? If so - we need to describe these services

Reply:
added links to docs for more info